examples/VisualNovel/Main.gd

```diff
@@ -0,0 +1,99 @@
+extends Control
+
+var transition_shader = preload("res://examples/VisualNovel/transition.gdshader")
+
+onready var background = $Background
+onready var background_transitions = $BackgroundTransitions
+onready var screencap = $Screencap
+onready var dialog_manager = $DialogNode.dialog_manager
+onready var portrait_manager = $DialogNode.portrait_manager
+onready var event_manager = $EventManager
+onready var ic_logs = $ICLogs
+
+var bgs = [
+	"res://examples/VisualNovel/Backgrounds/backgroundColorForest.png",
+	"res://examples/VisualNovel/Backgrounds/backgroundColorGrass.png",
+]
+var bg_index = 0
+
+var current_portrait: Portrait
+var current_event: Event
+var seen_events: Dictionary = {}
+
+var skip_timer = 0.0
+
+export var skip_delay: float = 0.02
+
+func _process(delta):
+	skip_timer += delta
+	var go = Input.is_action_just_pressed("ui_accept") or Input.is_action_pressed("ui_right")
+	if go and skip_timer > skip_delay:
+		skip_timer = 0.0
+		if not current_event:
+			event_manager.go_to_next_event()
+		elif event_manager.timeline in seen_events and \
+			 current_event in seen_events[event_manager.timeline]:
+				dialog_manager.display_all_text()
+
+	if Input.is_action_just_pressed("ui_down"):
+		bg_index = (bg_index + 1) % bgs.size()
+		set_background(bgs[bg_index], 0, 0)
+
+
+func set_background(bg_path: String, smooth: float = 0.5, speed: float = 1.0):
```

Comment:
an event for this function could be a good idea?

Reply:
Very true! I was considering creating events from the example itself, but it would defo be better to just make it part of Textalog.

Reply:
You can make it an event of textalog, adding the event to events folder and its path reference to the plugin script.